packages/cli/src/verify.ts

```diff
@@ -73,94 +74,114 @@ export async function verify({
     ),
   );
 
-  Object.entries(
-    usesProxy ? getWorldProxyFactoryContracts(deployerAddress) : getWorldFactoryContracts(deployerAddress),
-  ).map(([name, { bytecode }]) =>
-    verifyQueue.add(() =>
-      verifyContract(
-        {
-          name,
-          rpc,
-          verifier,
-          verifierUrl,
-          address: getCreate2Address({
-            from: deployerAddress,
-            bytecode: bytecode,
-            salt,
-          }),
-        },
-        {
-          profile: foundryProfile,
-          cwd: "node_modules/@latticexyz/world",
-        },
-      ).catch((error) => {
-        console.error(`Error verifying world factory contract ${name}:`, error);
-      }),
-    ),
-  );
+  {
+    const cwd = "node_modules/@latticexyz/world";
 
-  modules.map(({ name, bytecode }) =>
-    verifyQueue.add(() =>
-      verifyContract(
-        {
-          name: name,
-          rpc,
-          verifier,
-          verifierUrl,
-          address: getCreate2Address({
-            from: deployerAddress,
-            bytecode: bytecode,
-            salt,
-          }),
-        },
-        {
-          profile: foundryProfile,
-          cwd: "node_modules/@latticexyz/world-modules",
-        },
-      ).catch((error) => {
-        console.error(`Error verifying module contract ${name}:`, error);
-      }),
-    ),
-  );
+    await execa("npm", ["install"], {
+      cwd,
+    });
 
-  // If the world was deployed as a Proxy, verify the proxy and implementation.
-  if (usesProxy) {
-    const implementationAddress = sliceHex(implementationStorage, -20);
-
-    verifyQueue.add(() =>
-      verifyContract(
-        { name: "WorldProxy", rpc, verifier, verifierUrl, address: worldAddress },
-        {
-          profile: foundryProfile,
-          cwd: "node_modules/@latticexyz/world",
-        },
-      ).catch((error) => {
-        console.error(`Error verifying WorldProxy contract:`, error);
-      }),
+    Object.entries(
+      usesProxy ? getWorldProxyFactoryContracts(deployerAddress) : getWorldFactoryContracts(deployerAddress),
+    ).map(([name, { bytecode }]) =>
+      verifyQueue.add(() =>
+        verifyContract(
+          {
+            name,
+            rpc,
+            verifier,
+            verifierUrl,
+            address: getCreate2Address({
+              from: deployerAddress,
+              bytecode: bytecode,
+              salt,
+            }),
+          },
+          {
+            profile: foundryProfile,
+            cwd,
+          },
```

Comment:
thinking foundry profile makes less sense here because users can't control it (i.e. our deps may not have a "holesky" profile that you have locally) and now wondering if we should even offer the foundry profile option and instead just require the CLI flags for the things we need from the foundry config